lib/semantics/symbol.h

```diff
@@ -211,20 +213,28 @@ class DerivedTypeDetails {
   // Returns the complete list of derived type parameter symbols in
   // the order in which their declarations appear in the derived type
   // definitions (parents first).
-  std::list<const Symbol *> OrderParameterDeclarations(const Symbol &) const;
+  SymbolList OrderParameterDeclarations(const Symbol &) const;
+
+  // Returns the complete list of derived type components in the order
+  // in which their declarations appear in the derived type definitions
+  // (parents first).  Parent components appear in the list immediately
+  // after the components that belong to them.
+  SymbolList OrderComponents(const Scope &) const;
+
+  const Symbol *GetParentComponent(const Scope &) const;
 
 private:
   // These are (1) the names of the derived type parameters in the order
   // in which they appear on the type definition statement(s), and (2) the
   // symbols that correspond to those names in the order in which their
   // declarations appear in the derived type definition(s).
   std::list<SourceName> paramNames_;
-  std::list<const Symbol *> paramDecls_;
-  // These are the declarations of the derived type's components in component
+  SymbolList paramDecls_;
+  // These are the names of the derived type's components in component
   // order.  A parent component, if any, appears first in this list.
-  std::list<const Symbol *> components_;
-  SourceName extends_;
+  std::list<SourceName> components_;
```

Comment:
`componentNames_` might be a better name for this field. In the functions `OrderComponents()` and `GetParentComponent()`, "component" means component symbol, not name.

Reply:
Sure.